CustomerContactBook/Services/CustomersService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using CustomerContactBook.Database;
+using CustomerContactBook.Database.Tables;
+using CustomerContactBook.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CustomerContactBook.Services
+{
+    public class CustomersService
+    {
+        private readonly ContactBookContext _context;
+
+        public CustomersService(ContactBookContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<CustomerModel>> GetCustomers()
+        {
+            var result =  await _context.Customers.ToListAsync();
+            return result.Select(toCustomerModel).ToList();
+        }
+
+        public async Task<CustomerModel> GetCustomer(long id)
+        {
+            var customer = await _context.Customers.FindAsync(id);
+
+            if (customer == null)
+            {
+                return null;
+            }
+
+            return toCustomerModel(customer);
+        }
+
+        public async Task<bool?> PutCustomer(long id, CustomerModel model)
+        {
+            var customer = toCustomer(model);
+            if (customer == null)
+            {
+                return null;
+            }
+            _context.Entry(customer).State = EntityState.Modified;
```

Comment:
it's really tricky thing! I like it!
But please try to avoid complex things in the real projects - usually the cost of maintaining the code is higher than the amount of resources you save (CPU/network). That is, the other person will spend more time trying to understand what is written here. And even worse - they will not understand and create a bug.